src/sources/datadog/agent/metrics.rs

```diff
@@ -182,7 +215,142 @@ fn decode_datadog_sketches(
     Ok(metrics)
 }
 
-fn decode_datadog_series(
+fn decode_datadog_series_v2(
+    body: Bytes,
+    api_key: Option<Arc<str>>,
+    schema_definition: &Arc<schema::Definition>,
+) -> Result<Vec<Event>, ErrorMessage> {
+    if body.is_empty() {
+        // The datadog agent may send an empty payload as a keep alive
+        debug!(
+            message = "Empty payload ignored.",
+            internal_log_rate_secs = 30
+        );
+        return Ok(Vec::new());
+    }
+
+    let metrics = decode_ddseries_v2(body, &api_key, schema_definition).map_err(|error| {
+        ErrorMessage::new(
+            StatusCode::UNPROCESSABLE_ENTITY,
+            format!("Error decoding Datadog sketch: {:?}", error),
+        )
+    })?;
+
+    emit!(EventsReceived {
+        byte_size: metrics.size_of(),
+        count: metrics.len(),
+    });
+
+    Ok(metrics)
+}
+
+pub(crate) fn decode_ddseries_v2(
+    frame: Bytes,
+    api_key: &Option<Arc<str>>,
+    schema_definition: &Arc<schema::Definition>,
+) -> crate::Result<Vec<Event>> {
+    let payload = MetricPayload::decode(frame)?;
+    let decoded_metrics: Vec<Event> = payload
+        .series
+        .into_iter()
+        .flat_map(|serie| {
+            let mut tags: BTreeMap<String, String> = serie
+                .tags
+                .iter()
+                .map(|tag| {
+                    let kv = tag.split_once(':').unwrap_or((tag, ""));
+                    (kv.0.trim().into(), kv.1.trim().into())
+                })
+                .collect();
+
+            serie.resources.into_iter().for_each(|r| {
+                // As per https://github.com/DataDog/datadog-agent/blob/a62ac9fb13e1e5060b89e731b8355b2b20a07c5b/pkg/serializer/internal/metrics/iterable_series.go#L180-L189
+                // the hostname can be found in MetricSeries::resources and that is the only value stored there.
+                if r.r#type.eq("host") {
+                    tags.insert(log_schema().host_key().to_string(), r.name);
+                } else {
+                    // But to avoid losing information if this situation changes, any other resource type/name will be saved in the tags map
+                    tags.insert(format!("resource.{}", r.r#type), r.name);
+                }
+            });
+            (!serie.source_type_name.is_empty())
+                .then(|| tags.insert("source_type_name".into(), serie.source_type_name));
+            // As per https://github.com/DataDog/datadog-agent/blob/a62ac9fb13e1e5060b89e731b8355b2b20a07c5b/pkg/serializer/internal/metrics/iterable_series.go#L224
+            // serie.unit is omitted
+            match metric_payload::MetricType::from_i32(serie.r#type) {
+                Some(metric_payload::MetricType::Count) => serie
+                    .points
+                    .iter()
+                    .map(|dd_point| {
+                        Metric::new(
```

Comment:
Would it be a good idea to include the parsing out of the namespace here in the v2 as was added recently for the v1?

https://github.com/vectordotdev/vector/blob/1df4d50ae20491c2a87e280aed7395e2ca79054b/src/sources/datadog/agent/metrics.rs#L413

(per this PR) : https://github.com/vectordotdev/vector/pull/13176/files

Reply:
Agreed, this got automerged, but we should add this namespace parsing here too.

Reply:
Hmmm, new issue then ? I can open it.

Reply:
I did it in #13376